addons/web/static/src/core/file_upload/file_upload_service.js

```diff
@@ -68,6 +68,18 @@ export const fileUploadService = {
             // Load listener
             xhr.addEventListener("load", () => {
                 delete uploads[upload.id];
+                if (xhr.readyState === 4 && xhr.status === 400) {
+                    const parser = new DOMParser();
+                    const doc = parser.parseFromString(xhr.responseText, "text/html");
+                    const pTag = doc.querySelector("p");
+                    const errorMessage = _t(
+                        "An error occurred while uploading %(fileName)s, %(errorMessage)s",
+                        { fileName: upload.title, errorMessage: pTag.textContent.trim() }
+                    );
```

Comment:
```suggestion
                    const errorMessage = pTag?.textContent?.trim() || _t("Unexpected error occurred while uploading.");
                    notificationService.add(errorMessage, {
                        sticky: true,
                        title: _t('An error occurred while uploading %(fileName)s', { fileName: upload.title }), 
                        type: "danger",
                    });
```
- Good to add a variable `errorMessage` and use pTag.textContent here, also sure textContent is present in pTag otherwise static message is good.
- Stick this message so users will completely read this messag.
- In error title should mention file name